### PR TITLE
LPS-172966 Avoid processing images when fileEntry is not an image

### PR DESCRIPTION
**Pull request by thienquach87**
Hi Team,

Here is the PR for fixing [LPS-172966](https://issues.liferay.com/browse/LPP-47675). [The original PR](https://github.com/thienquach87/liferay-portal/pull/26) was passed my review.
Please help review and leave your comments.
#cc: @tototrinh

Thanks,
Thien

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/adaptive-media/adaptive-media-image-content-transformer-backwards-compatibility/src/main/java/com/liferay/adaptive/media/image/content/transformer/backwards/compatibility/internal/AMBackwardsCompatibilityHtmlContentTransformer.java [hunk 1]
```diff
@@ -143,7 +144,9 @@ protected Pattern getPattern() {
 	protected String getReplacement(String originalImgTag, FileEntry fileEntry)
 		throws PortalException {
 
-		if (fileEntry == null) {
+		if ((fileEntry == null) ||
+			!ImageProcessorUtil.isSupported(fileEntry.getMimeType())) {
```
Comment (adolfopa) [inline modules/apps/adaptive-media/adaptive-media-image-content-transformer-backwards-compatibility/src/main/java/com/liferay/adaptive/media/image/content/transformer/backwards/compatibility/internal/AMBackwardsCompatibilityHtmlContentTransformer.java:148]: This is adaptive media; you should use `AMImageMimeTypeProvider` to validate the mime types.  The set of mime types supported by adaptive media are not necessarily the same as those of the image processor.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/adaptive-media/adaptive-media-image-content-transformer-backwards-compatibility/src/main/java/com/liferay/adaptive/media/image/content/transformer/backwards/compatibility/internal/AMBackwardsCompatibilityHtmlContentTransformer.java (hunk 1 above)
Comment (adolfopa) [reply modules/apps/adaptive-media/adaptive-media-image-content-transformer-backwards-compatibility/src/main/java/com/liferay/adaptive/media/image/content/transformer/backwards/compatibility/internal/AMBackwardsCompatibilityHtmlContentTransformer.java:148]: See also `AMImageValidator.isProcessingSupported()`. That's probably what you want to use here.

Comment (tototrinh) [reply modules/apps/adaptive-media/adaptive-media-image-content-transformer-backwards-compatibility/src/main/java/com/liferay/adaptive/media/image/content/transformer/backwards/compatibility/internal/AMBackwardsCompatibilityHtmlContentTransformer.java]: Thanks @adolfopa for your feedback, I updated the code. Please help take a look again. Thanks.

**Comment by tototrinh** [comment]
ci:test:sf

**Comment by tototrinh** [comment]
ci:test:relevant

File: modules/apps/adaptive-media/adaptive-media-image-content-transformer-backwards-compatibility/src/main/java/com/liferay/adaptive/media/image/content/transformer/backwards/compatibility/internal/AMBackwardsCompatibilityHtmlContentTransformer.java (hunk 1 above)
Comment (tototrinh) [reply modules/apps/adaptive-media/adaptive-media-image-content-transformer-backwards-compatibility/src/main/java/com/liferay/adaptive/media/image/content/transformer/backwards/compatibility/internal/AMBackwardsCompatibilityHtmlContentTransformer.java]: `AMImageValidator.isProcessingSupported()` will return true for non-image files so the issue still occurs. So I use `AMImageMimeTypeProvider.isMimeTypeSupported` instead.

**Review by adolfopa — APPROVED** [review]

**Pull request closed without merging**